src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/SmeltersTool.java

```diff
@@ -0,0 +1,97 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.tools;
+
+import io.github.thebusybiscuit.slimefun4.api.MinecraftVersion;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
+import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.attributes.DamageableItem;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ToolUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import io.github.thebusybiscuit.slimefun4.utils.tags.SlimefunTag;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import org.bukkit.Effect;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+public class SmeltersTool extends SimpleSlimefunItem<ToolUseHandler> implements DamageableItem {
+
+    @ParametersAreNonnullByDefault
+    public SmeltersTool(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(itemGroup, item, recipeType, recipe);
+    }
+
+    @Override
+    public @Nonnull
+    ToolUseHandler getItemHandler() {
+        return (e, tool, fortune, drops) -> {
+            Block b = e.getBlock();
+            ItemStack stack = e.getPlayer().getInventory().getItemInMainHand();
+
+            if (SlimefunTag.PICKAXES.isTagged(stack.getType()) && SlimefunTag.SMELTERS_PICKAXE_BLOCKS.isTagged(b.getType()) && !BlockStorage.hasBlockInfo(b)) {
+                Collection<ItemStack> blockDrops = b.getDrops(tool);
```

Comment:
this part is kinda redundant.

Reply:
let it be as long as its doesnt break anything

Reply:
ybw is right, this is only used the once, so you can remove this and do `for (ItemStack drop : b.getDrops(tool)) {`

Reply:
eh, actually I mean the whole if block, it repeats 3 times.

Reply:
maybe i should select more lines next time